### PR TITLE
CBG-536 & CBG-537 : Fix for data race in Webhook test handlers and  intermittent assertion failures

### PR DESCRIPTION
**Pull request by sarathkumarsivan**
CBG-537: Fix for data race in Webhook test handlers

### CODE REVIEW

File: db/event_manager_test.go [hunk 1]
```diff
@@ -2,6 +2,8 @@ package db
 
 import (
 	"fmt"
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/stretchr/testify/assert"
```
Comment (adamcfraser) [inline db/event_manager_test.go]: The standard practice with go is to separate the standard library imports from non-standard library imports, with an empty line between the two sets.

File: db/event_manager_test.go [hunk 2]
```diff
@@ -39,26 +38,26 @@ func (th *TestingHandler) HandleEvent(event Event) {
...
-		goassert.True(th.t, state != nil && (state == "online" || state == "offline"))
+		assert.True(th.t, state != nil && (state == "online" || state == "offline"))
 
 		//admin interface must resolve to a a valis tcp address
 		adminInterface := (doc["admininterface"]).(string)
 
 		_, err := net.ResolveTCPAddr("tcp", adminInterface)
 
-		goassert.Equals(th.t, err, nil)
+		assert.Equal(th.t, nil, err)
```
Comment (adamcfraser) [inline db/event_manager_test.go]: assert.NoError is preferable here and below.

File: db/event_manager_test.go [hunk 3]
```diff
@@ -104,7 +103,7 @@ func TestDocumentChangeEvent(t *testing.T) {
 	//Raise events
 	for i := 0; i < 10; i++ {
 		body, docid, channels := eventForTest(i)
-		bodyBytes, _ := base.JSONMarshal(body)
+		bodyBytes, _ := base.JSONMarshalCanonical(body)
```
Comment (adamcfraser) [inline db/event_manager_test.go]: Is it necessary to change all JSONMarshal usages in event_maanger_test?  It doesn't look like all tests are trying to assert the event body.

**Review by adamcfraser — CHANGES_REQUESTED** [review]

File: db/event_manager_test.go (hunk 2 above)
Comment (sarathkumarsivan) [reply db/event_manager_test.go]: Thank you Adam. This has been corrected.

File: db/event_manager_test.go (hunk 1 above)
Comment (sarathkumarsivan) [reply db/event_manager_test.go]: Thank you Adam, I did a quick formatting on the entire file initially. It is fixed as per the standard.

File: db/event_manager_test.go (hunk 3 above)
Comment (sarathkumarsivan) [reply db/event_manager_test.go]: Yeah, you are right it's not required in all tests. The change has been reverted and corrected in the appropriate place.

**Review by bbrks — APPROVED** [review]
LGTM

**Review by adamcfraser — APPROVED** [review]

**Pull request merged**